### ISSUE TITLE
FormLayout#addFormItem label click doesn't focus/toggle checkbox

### ISSUE THREAD

**Issue by leluna** [vaadin/flow-components#5110]
### Describe your motivation

When using FormLayout with the label on the left, a `<label>` element is created by the method `FormLayout#addFormItem`.  However, as web components are not form-associated, the label element does not really work as they should, e.g. click events on the label for a Vaadin check box does not trigger the checkbox.

Is there any reason why not to set form-associated to true?

### Describe the solution you'd like

I would expect the labels in a `FormLayout` to be really associated to the component

**Comment by web-padawan** [comment]
Thanks for the issue. We do not handle `for` attribute on the native `<label>`, although we probably could.
There is some logic to link this label to the `<input>` via `aria-labelledby` attribute for screen readers.

**Comment by leluna** [comment]
According to the spec https://html.spec.whatwg.org/#category-label, labels needs to be used with a form-associated custom element. Technically, the current state is spec conform I guess, as the `for` attribute is not even set. So why is `FormLayout#addFormItem` using `<label>` at all then? Wouldn't a `span` suffice?

**Comment by web-padawan** [comment]
> So why is FormLayout#addFormItem using <label> at all then? Wouldn't a span suffice?

Here is the screenshot illustrating that `<label>` is linked through `aria-labelledby` attribute for screen readers.

<img width="522" alt="Screenshot 2023-06-02 at 10 52 30" src="https://github.com/vaadin/flow-components/assets/10589913/ee3684a2-9a13-40ad-8a38-7d00eba518a9">

Apart from that, we do have [custom logic](https://github.com/vaadin/web-components/blob/1abaa28999c80a0511af0ce73ad1f73592579a5a/packages/form-layout/src/vaadin-form-item.js#L220-L226) in the web component ensuring that clicking the label focuses the field. 
As far as I understand, the only missing piece in fact is `for` attribute. We could consider adding logic for it.

**Comment by knoobie** [comment]
Isn't in your screenshot a `for` attribute linked to the input? :)

**Comment by web-padawan** [comment]
> Isn't in your screenshot a for attribute linked to the input? :)

No. The actual label of the `vaadin-form-item` is the one that has `id="label-vaadin-form-item-62"`
You can check it in the [Label Position](https://vaadin.com/docs/latest/components/form-layout/#label-position) docs example - the screenshot above is from there.

**Comment by knoobie** [comment]
Thanks! My brain stopped scanning for label once I saw the first one. Haven't noticed this in our apps, but I remembered, we switched from #addFormItem to the normal add and only apply labels directly to the fields after our last refactoring.

**Comment by web-padawan** [comment]
The reason why clicking `vaadin-form-item` label didn't fo toggle cus / checkbox was that the logic for getting the [field node](https://github.com/vaadin/web-components/blob/badc59aa4318652c88f1a467c192d715e3adb38c/packages/form-layout/src/vaadin-form-item.js#L299-L301) was expecting `validate()` or `checkValidity()` method and back then, the checkbox didn't provide them. This was fixed byadding validation API to `vaadin-checkbox` in https://github.com/vaadin/web-components/pull/7285 which landed in Vaadin 24.4. Closing.
